rocketpy/motors/fluid.py

```diff
@@ -1,44 +1,124 @@
+        ----------
+        name : str
+            Name of the fluid.
+        density : int, float, callable, string, array, Function
+            Density of the fluid in kg/m³ as a function of temperature
+            and pressure. If a int or float is given, it is considered
+            a constant function. A callable, csv file or an list of points
+            can be given to express the density values.
+            The parameter is used as a ``Function`` source.
+
+            See more on :class:`rocketpy.Function` source types.
         """
 
-        if not isinstance(self.name, str):  # pragma: no cover
-            raise ValueError("The name must be a string.")
-        if self.density < 0:  # pragma: no cover
-            raise ValueError("The density must be a positive number.")
+        self.name = name
+        self.density = density
+        self.density_function = density
 
-        # Initialize plots and prints object
         self.prints = _FluidPrints(self)
         self.plots = _FluidPlots(self)
 
+    @property
+    def density_function(self):
+        """Density of the fluid as a function of temperature and pressure."""
+        return self._density_function
+
+    @density_function.setter
+    def density_function(self, value):
+        """Setter for the density function of the fluid. Numeric values
+        are converted to constant functions.
+
+        Parameters
+        ----------
+        value : int, float, callable, string, array, Function
+            Density of the fluid in kg/m³ as a function of temperature
+            and pressure. The value is used as a ``Function`` source.
+        """
+        if isinstance(value, (int, float, np.number)):
+            # pylint: disable=unused-argument
+            def density_function(temperature, pressure):
+                return value
+
+        else:
+            density_function = value
+
+        self._density_function = Function(
+            density_function,
+            interpolation="linear",
+            extrapolation="natural",
+            inputs=["Temperature (K)", "Pressure (Pa)"],
+            outputs="Density (kg/m³)",
+        )
+
+    def get_time_variable_density(self, temperature, pressure):
+        """Get the density of the fluid as a function of time.
+
+        Parameters
+        ----------
+        temperature : Function
+            Temperature of the fluid in Kelvin.
+        pressure : Function
+            Pressure of the fluid in Pascals.
+
+        Returns
+        -------
+        Function
+            Density of the fluid in kg/m³ as function of time.
+        """
+        if callable(temperature.source) and callable(pressure.source):
+            return Function(
+                lambda time: self.density_function.get_value(
+                    temperature.source(time), pressure.source(time)
+                ),
+                inputs="Time (s)",
+                outputs="Density (kg/m³)",
+            )
+        else:
```

Comment:
You don't need to use the "else" here. There's a return within the "if" block.

```suggestion
```

Reply:
Solved in [568661d](https://github.com/RocketPy-Team/RocketPy/pull/852/commits/568661d1e42253d1a0fc2b62a27858e7d71f1880).